cli/cmd/sync_v3.go

```diff
@@ -345,52 +345,67 @@ func syncConnectionV3(ctx context.Context, source v3source, destinations []v3des
 			return fmt.Errorf("unknown message type: %T", m)
 		}
 	}
+
 	err = syncClient.CloseSend()
 	if err != nil {
 		return err
 	}
 	totals := sourceClient.Metrics()
-
-	for i := range destinationsClients {
-		if destinationSpecs[i].WriteMode == specs.WriteModeOverwriteDeleteStale {
-			if err := deleteStale(writeClients[i], tablesForDeleteStale, sourceName, syncTime); err != nil {
-				return err
-			}
-		}
-		if _, err := writeClients[i].CloseAndRecv(); err != nil {
-			return err
-		}
-		if _, err := destinationsPbClients[i].Close(ctx, &plugin.Close_Request{}); err != nil {
-			return err
-		}
-	}
-
 	sourceWarnings := totals.Warnings
 	sourceErrors := totals.Errors
-	syncSummaries := make([]syncSummary, len(destinationsClients))
+	var metadataDataErrors error
 	for i := range destinationsClients {
 		m := destinationsClients[i].Metrics()
-		totals.Warnings += m.Warnings
-		totals.Errors += m.Errors
-		syncSummaries[i] = syncSummary{
+		summary := syncSummary{
 			Resources:           uint64(totalResources),
 			SourceErrors:        sourceErrors,
 			SourceWarnings:      sourceWarnings,
 			SyncID:              uid,
+			SyncTime:            syncTime,
 			SourceName:          sourceSpec.Name,
 			SourceVersion:       sourceSpec.Version,
 			SourcePath:          sourceSpec.Path,
-			CliVersion:          Version,
+			CLIVersion:          Version,
 			DestinationErrors:   m.Errors,
 			DestinationWarnings: m.Warnings,
 			DestinationName:     destinationSpecs[i].Name,
 			DestinationVersion:  destinationSpecs[i].Version,
 			DestinationPath:     destinationSpecs[i].Path,
 		}
+
+		if err := persistSummary(summaryLocation, summary); err != nil {
+			log.Warn().Err(err).Msg("Failed to persist sync summary")
+		}
+
+		log.Info().Interface("summary", summary).Msg("Sync summary")
+		if !destinationSpecs[i].SyncSummary {
+			continue
+		}
+		// Only send the summary to the destination that matches the current destination
+		if destinationSpecs[i].Name != summary.DestinationName || destinationSpecs[i].Version != summary.DestinationVersion || destinationSpecs[i].Path != summary.DestinationPath {
+			continue
+		}
+		if err := sendSummary(writeClients[i], destinationSpecs[i], destinationsClients[i], destinationTransformers[i], &summary, noMigrate); err != nil {
+			metadataDataErrors = errors.Join(metadataDataErrors, err)
+		}
 	}
-	err = persistSummary(summaryLocation, syncSummaries)
-	if err != nil {
-		log.Warn().Err(err).Msg("Failed to persist sync summary")
+	// a single error for one of the destination should not stop all of the other table from being written
```

Comment:
```suggestion
```

I would say remove this comment or move it above `metadataDataErrors = errors.Join(metadataDataErrors, err)`